server/consumer.go

```diff
@@ -1677,6 +1677,11 @@ func (o *consumer) processNextMsgReq(_ *subscription, c *client, _, reply string
 		return
 	}
 
+	if o.maxp > 0 && batchSize > o.maxp {
```

Comment:
I think you want to look at current pending + batchSize and if that is > o.maxp no?

Reply:
Not sure... That seems to change the behavior of redeliveries, gave it a try but cannot get `TestJetStreamPullConsumerMaxAckPendingRedeliveries` to pass since think that would mean that having too many unacked inflight messages prevent making pull requests:

https://github.com/nats-io/nats-server/compare/master...wallyqs:consumer-ack-pending-pull-cap